### DISCUSSION TITLE
求助请问哪位帮我看看哪里出了问题

### DISCUSSION THREAD

**Discussion by toyqiu** [NewFuture/DDNS#236] [Q&A]
docker运行的

config.json文件如下
```json
{
  "$schema": "https://ddns.newfuture.cc/schema/v2.8.json",
  "id": "***",
  "token": "***",
  "dns": "dnspod",
  "ipv4": ["125.43.68.66"],
  "ipv6": ["ooxx.cc"],
  "index4": 0,
  "index6": "public",
  "ttl": 600,
  "proxy": "DIRECT",
  "debug": false
}
```

日志如下
![image](https://user-images.githubusercontent.com/479873/111100472-d292a280-8582-11eb-9e78-5973495a469c.png)

但是就是不更新呀，请问是咋回事

**Comment by NN708** [comment]
配置文件 `"ipv4"` 这里填写的应该是你希望更新的 A 记录的域名，而不是你的 IP 地址。另外，将 id 和 token 的明文直接发在 issue 里不安全，建议删除。

**Comment by toyqiu** [comment]
> 配置文件 `"ipv4"` 这里填写的应该是你希望更新的 A 记录的域名，而不是你的 IP 地址。另外，将 id 和 token 的明文直接发在 issue 里不安全，建议删除。

谢谢谢谢，我再试试
id  token改过了

**Comment by toyqiu** [comment]
> 配置文件 `"ipv4"` 这里填写的应该是你希望更新的 A 记录的域名，而不是你的 IP 地址。另外，将 id 和 token 的明文直接发在 issue 里不安全，建议删除。

您好，我改成如下情况之后发现公网ip变了之后域名直接指向127.0.0.1了


![image](https://user-images.githubusercontent.com/479873/111106791-b85fc100-8590-11eb-9a66-b335c78656a2.png)


日志部分如下

![image](https://user-images.githubusercontent.com/479873/111106884-d7f6e980-8590-11eb-8ac5-ce14feed24a1.png)

**Comment by NN708** [comment]
你的 0 号网卡 IP 地址是 127.0.0.1。请参考 [index4 和 index6 参数说明](https://github.com/NewFuture/DDNS#index4-和-index6-参数说明)，按照你的需要正确配置 `"index4"` 参数。

**Comment by toyqiu** [comment]
> 你的 0 号网卡 IP 地址是 127.0.0.1。请参考 [index4 和 index6 参数说明](https://github.com/NewFuture/DDNS#index4-%E5%92%8C-index6-%E5%8F%82%E6%95%B0%E8%AF%B4%E6%98%8E)，按照你的需要正确配置 `"index4"` 参数。

不好意思呀，我不太理解这个地方填什么，是要填几号网卡吗？群晖上只有一个网卡呀

**Comment by NN708** [comment]
若没有特殊需求，请使用 "default"；如果是需要公网 IP，请使用 "public"。

**Comment by toyqiu** [comment]
> 若没有特殊需求，请使用 "default"；如果是需要公网 IP，请使用 "public"。

谢谢谢谢！
如果default的话会有什么效果呀？

**Comment by NN708** [comment]
> > 若没有特殊需求，请使用 "default"；如果是需要公网 IP，请使用 "public"。
> 
> 谢谢谢谢！
> 如果default的话会有什么效果呀？

得到你设备上网卡的地址。如果你的设备在局域网内，一般就是其内网地址。

**Comment by toyqiu** [comment]
> > > 若没有特殊需求，请使用 "default"；如果是需要公网 IP，请使用 "public"。
> > 
> > 
> > 谢谢谢谢！
> > 如果default的话会有什么效果呀？
> 
> 得到你设备上网卡的地址。如果你的设备在局域网内，一般就是其内网地址。

我的目的就是要dnspod更新家宽公网，所以是用public哦？
感谢！！！！

**Comment by NN708** [comment]
> > > > 若没有特殊需求，请使用 "default"；如果是需要公网 IP，请使用 "public"。
> > > 
> > > 
> > > 谢谢谢谢！
> > > 如果default的话会有什么效果呀？
> > 
> > 
> > 得到你设备上网卡的地址。如果你的设备在局域网内，一般就是其内网地址。
> 
> 我的目的就是要dnspod更新家宽公网，所以是用public哦？
> 感谢！！！！

是的
